src/compiler/factory/nodeFactory.ts

```diff
@@ -6839,6 +6840,7 @@ namespace ts {
             }
         }
         if (startsOnNewLine !== undefined) destEmitNode.startsOnNewLine = startsOnNewLine;
+        if (snippetElement !== undefined) destEmitNode.snippetElement = snippetElement;

```

Comment:
What exactly does the `snippetElement` property do? I didn't really see need this change on my branch.

Reply:
Oh I guess this is for placing the cursor in the right place - but is there an easy way to show this not working right now?

Reply:
It needs to clone this snippet. We need this when we use _snippets_
https://github.com/microsoft/TypeScript/blob/71d19115031ae1c48fa34a24636dfae5e87e1b08/src/services/completions.ts#L937

Reply:
In the current implementation we don’t use clone for `body`

Reply:
If we try to clone the body without these changes, we lose the snippet - for example, tests with enabled `includeCompletionsWithSnippetText` will fail (`completionsOverridingMethod2`, etc.)